pkg/search/postgres/store.go

```diff
@@ -108,7 +108,7 @@ func (s *GenericStore[T, PT]) Exists(ctx context.Context, id string) (bool, erro
 		if ok, err := s.permissionChecker.ExistsAllowed(ctx); err != nil {
 			return false, err
 		} else if !ok {
-			return false, nil
+			return false, sac.ErrResourceAccessDenied
```

Comment:
It looks like this PR was missed by re-stack as it brings back behaviours for which changes were requested.
Please make sure it only has the minimal change set.